pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,37 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
+		req.BucketName = obj.Spec.Parameters["BucketInstanceName"]
+	} else if len(obj.Spec.Parameters["BucketPrefix"]) > 0 {
+		req.BucketContext["BucketPrefix"] = obj.Spec.Parameters["BucketPrefix"]
+	}
+
+	if obj.Spec.AnonymousAccessMode.Private {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_PRIVATE
+	} else if obj.Spec.AnonymousAccessMode.PublicReadOnly {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_READ_ONLY
+	} else if obj.Spec.AnonymousAccessMode.PublicReadWrite {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_WRITE_ONLY
+	} else if obj.Spec.AnonymousAccessMode.PublicWriteOnly {
+		req.AnonymousBucketAccessMode = osspec.ProvisionerCreateBucketRequest_BUCKET_READ_WRITE
 	}
 
 	switch obj.Spec.Protocol.Name {
 	case v1alpha1.ProtocolNameS3:
 		req.Region = obj.Spec.Protocol.S3.Region
+		req.BucketContext["Version"] = obj.Spec.Protocol.S3.Version
```

Comment:
do we want to use hard-coded strings for keys, or is it better to define constants for these values so that, if we ever change them, we need only change the one const?